extensions/mesa-vpu.sh

```diff
@@ -0,0 +1,110 @@
+#
+# SPDX-License-Identifier: GPL-2.0
+# Armbian build framework extension
+#
+# Enables 3D and multimedia, 4K VPU with Chromium, acceleration for Ubuntu. Debian only 3D
+#
+
+function extension_prepare_config__3d() {
+
+	# only used when generating image
+	[[ "${BUILDING_IMAGE}" != "yes" ]] && return 0
+
+	# only used when generating desktop
+	[[ "${BUILD_DESKTOP}" != "yes" ]] && return 0
+
+	# Define image suffix
+	if [[ "${LINUXFAMILY}" =~ ^(rockchip-rk3588|rk35xx)$ && "$BRANCH" =~ ^(legacy)$ && "${RELEASE}" =~ ^(jammy|noble)$ ]]; then
+
+		EXTRA_IMAGE_SUFFIXES+=("-panfork")
+
+	elif [[ "${DISTRIBUTION}" == "Ubuntu" ]]; then
+
+		EXTRA_IMAGE_SUFFIXES+=("-oibaf")
+
+	fi
+
+        # This should be enabled on all for rk3588 distributions where mesa and vendor kernel is present
+        if [[ "${LINUXFAMILY}" =~ ^(rockchip-rk3588|rk35xx)$ && "$BRANCH" == vendor ]]; then
+                declare -g DEFAULT_OVERLAYS="panthor-gpu"
+        fi
+
+}
+
+function post_install_kernel_debs__3d() {
+
+	# Silently deny old releases which are not supported but are still in the system
+	[[ "${RELEASE}" =~ ^(bullseye|buster|focal)$ ]] && return 0
+
+	# Do not install those packages on CLI and minimal images
+	[[ "${BUILD_DESKTOP}" != "yes" ]] && return 0
+
+	# Packages that are going to be installed
+	declare -a pkgs=("mesa-utils" "mesa-utils-extra" "libglx-mesa0" "libgl1-mesa-dri" "glmark2" "glmark2-wayland" "glmark2-es2-wayland" "glmark2-es2")
+
+	# Some packages, x11gl benchmark, came late into Ubuntu
+	[[ "${RELEASE}" != jammy ]] && pkgs+=("glmark2-x11" "glmark2-es2-x11")
+
+	# Rockchip RK3588 will use panfork only with legacy kernel
+	if [[ "${LINUXFAMILY}" =~ ^(rockchip-rk3588|rk35xx)$ && "$BRANCH" =~ ^(legacy)$ && "${RELEASE}" =~ ^(jammy|noble)$ ]]; then
+
+		EXTRA_IMAGE_SUFFIXES+=("-panfork") # Add to the image suffix. # global array
```

Comment:
Do we need this  since it's already declared in `extension_prepare_config__3d()`?